plugins/test/framework.cc

```diff
@@ -193,7 +195,9 @@ TestHttpContext::Result TestHttpContext::SendRequestBody(std::string body) {
   current_callback_ = TestHttpContext::CallbackType::RequestBody;
   result_.body_status =
       onRequestBody(body_buffer_.size(), /*end_of_stream=*/false);
-  result_.body = body_buffer_.release();
+  if (!sent_local_response_) {
```

Comment:
I wonder a little whether this should return BadArgument if sent_local_response_ is already true. Though that check is not currently performed in TuringStreamContext either.